frameworks/bitcoin-move/sources/bbn.move

```diff
@@ -179,77 +225,198 @@ module bitcoin_move::bbn {
-            if (script_buf::bytes(txout_script_pubkey(tx_output))
-                == staking_output_script) {
-                let out_point = types::new_outpoint(
-                    tx_id, (txout_value(tx_output) as u32)
-                );
-                return (true, index, option::some(utxo::derive_utxo_id(out_point)))
-            };
-            index = index + 1;
-        };
-        return (false, index, none())
+    public entry fun process_bbn_tx_entry(txid: address){
+        process_bbn_tx(txid)
     }
 
-    public fun derive_bbn_utxo(utxo_obj: &Object<UTXO>) {
-        // assert!(object::owner(utxo_obj) == @bitcoin_move, ErrorNotBabylonUTXO);
-        let utxo = object::borrow(utxo_obj);
-        let txid = utxo::txid(utxo);
-        let option_tx = bitcoin::get_tx(txid);
-        assert!(is_some(&option_tx), ErrorNotTransaction);
-        let transaction = option::destroy_some(option_tx);
+    fun process_bbn_tx(txid: address) {
+        let block_height_opt = bitcoin::get_tx_height(txid);
+        assert!(is_some(&block_height_opt), ErrorTransactionNotFound);
+        let block_height = option::destroy_some(block_height_opt);
+        let tx_opt = bitcoin::get_tx(txid);
+        assert!(is_some(&tx_opt), ErrorTransactionNotFound);
+        let transaction = option::destroy_some(tx_opt);
+
         let (is_true, op_return_index, op_return_data) =
             try_get_bbn_op_return_data(transaction);
+        
         assert!(is_true, ErrorNotBabylonOpReturn);
-        // TODO here should replace to check staking output
-        // try_get_staking_output()
+        
+        assert!(tx_lock_time(&transaction) >= (op_return_data.staking_time as u32), ErrorTransactionLockTime);
 
-        assert!(
-            tx_lock_time(&transaction) >= (op_return_data.staking_time as u32),
-            ErrorTransactionLockTime
-        );
+        let seal_protocol = type_info::type_name<BBNStakeSeal>();
         let tx_outputs = tx_output(&transaction);
         let index = 0;
-        // TODO bbn should not multiple staking outputs, we temporarily support
+        let has_stake_seal = false;
+        //bbn should not multiple staking outputs yet, we support it for in case
         while (index < length(tx_outputs)) {
             let tx_output = borrow(tx_outputs, index);
             if (index == op_return_index) {
                 continue
             };
-            let out_point = types::new_outpoint(txid, (txout_value(tx_output) as u32));
-            let borrow_utxo = utxo::borrow_utxo(out_point);
-            if (object::owner(borrow_utxo) != @bitcoin_move) {
+            let vout = (index as u32);
+            let txout_value = txout_value(tx_output);
+            let out_point = types::new_outpoint(txid, vout);
+            let utxo_obj = utxo::borrow_mut_utxo(out_point);
+            let utxo = object::borrow_mut(utxo_obj);
+            if (utxo::has_seal_internal(utxo, &seal_protocol)){
```

Comment:
Why is seal used here instead of temp state?

Reply:
1. We can display the seal of the UTXO more easily than the temp state.
2. The data is already in the BBNStakeSeel Object; we only need to store the ID.
3. In the future, we can destroy the BBNStakeSeel Object when the UTXO is spent.